src/sentry/workflow_engine/processors/data_source.py

```diff
@@ -0,0 +1,36 @@
+import sentry_sdk
+from django.db.models import Prefetch
+
+from sentry.utils import metrics
+from sentry.workflow_engine.models import DataPacket, DataSource, Detector
+
+
+def process_data_sources(
+    data_packets: list[DataPacket], query_type: DataSource.Type = DataSource.Type.SNUBA_QUERY
```

Comment:
What will pass in the query type here? So say we have a metric alert, is the idea that the kafka consumer will pass data like `process_data_sources([subscription_result], DataSource.Type.SNUBA_QUERY)`?

Reply:
yeah, that's what i was thinking. this type is more specific to the data itself, where the `detector` might have a type more specific to the product line.